dynamiqs/solvers/utils/batching.py

```diff
@@ -1,20 +1,89 @@
+from typing import List
+
+import torch
 from torch import Tensor
 
+from ...utils.tensor_types import ArrayLike, to_tensor
 from .td_tensor import TDTensor
 
 
 def batch_H(H: TDTensor) -> TDTensor:
     # handle H batching
-    # x: (b_H?, n, n) ->  (b_H, 1, n, n)
+    # x: (b_H?, n, n) ->  (b_H, 1, 1, n, n)
     if H.dim() == 2:  # (n, n)
         H = H.unsqueeze(0)  # (1, n, n)
-    return H.unsqueeze(1)
+    H = H.unsqueeze(1)
+    H = H.unsqueeze(1)
+    return H
+
+
+def batch_jump_ops(jump_ops: List[ArrayLike], dtype=None, device=None) -> Tensor:
```

Comment:
Use `list` instead of `List`, and `from __future__ import annotations` at the top of the file to use new-style typing. Pls also type `dtype` and `device`.

Reply:
We need a coding guidelines document to write down all these things